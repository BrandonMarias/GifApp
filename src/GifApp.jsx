import { useState } from "react";
import {GifGrid,AddCategory} from "./components";

export const GifApp = () => {
    const [categories, setCategories] = useState([]);

    const handleAddCategory = (newCategory = "") => {
        newCategory = newCategory.toLocaleLowerCase();
        if (categories.includes(newCategory)) return;
        setCategories((cat) => [newCategory, ...cat]);
    };

    return (
        <>
            <h1>Gif App</h1>
            <AddCategory handleFunction={handleAddCategory} />

            {categories.map((category) => (
                <GifGrid key={category} category={category} />
            ))}
        </>
    );
};
