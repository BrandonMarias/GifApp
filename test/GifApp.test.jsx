import { GifApp } from "../src/GifApp";
import { screen, render, fireEvent, waitFor } from "@testing-library/react";

describe("test on <GifApp/>", () => {
    test("should render correctly", () => {
        const { container } = render(<GifApp />);
        expect(container).toMatchSnapshot();
    });

    test("should add a new category", async () => {
        render(<GifApp />);
        const input = screen.getByRole("textbox");
        const form = screen.getByRole("form");

        fireEvent.input(input, { target: { value: "one punch" } });
        fireEvent.submit(form);
        fireEvent.input(input, { target: { value: "goku" } });
        fireEvent.submit(form);

        await waitFor(() => {
            expect(screen.getAllByRole("img")).toHaveLength(10);
        });
    });

    test("should not add a new category if it is repeated", async () => {
        render(<GifApp />);
        const input = screen.getByRole("textbox");
        const form = screen.getByRole("form");

        fireEvent.input(input, { target: { value: "one punch" } });
        fireEvent.submit(form);
        fireEvent.input(input, { target: { value: "one punch" } });
        fireEvent.submit(form);

        await waitFor(() => {
            expect(screen.getAllByRole("img")).toHaveLength(5);
        });
    });
});
