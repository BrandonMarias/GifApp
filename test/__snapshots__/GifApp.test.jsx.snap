// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifApp/> should render correctly 1`] = `
<div>
  <h1>
    Gif App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Bucar Gifs"
      type="text"
      value="one punch"
    />
  </form>
</div>
`;
